src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -1078,6 +1080,60 @@ class spell_warr_heroic_leap_jump : public SpellScriptLoader
     }
 };
 
+// Shockwave - 46968
+class spell_warr_shockwave : public SpellScriptLoader
+{
+public:
+	spell_warr_shockwave() : SpellScriptLoader("spell_warr_shockwave") { }
+
+	class spell_warr_shockwave_SpellScript : public SpellScript
+	{
+		PrepareSpellScript(spell_warr_shockwave_SpellScript);
+
+		int32 hitTargets = 0;
+		
+		bool Validate(SpellInfo const* /*spellInfo*/) override
+		{
+			if (!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_SHOCKWAVE) ||
+				!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_SHOCKWAVE_STUN))
```

Comment:
dosnt that mean it only counts if you stun and hit? because on retail you dont have to stun to get the reduction. just hit 3+ targets even if its 3 units with stunn immunity. and the dmg is dealt regardles if it stunned or not.

sorry if im wrong here